protobuf/src/reflect/types.rs

```diff
@@ -222,6 +222,10 @@ impl ProtobufType for ProtobufTypeInt32 {
     }
 
     fn compute_size(value: &i32) -> u32 {
+        // See also: https://github.com/protocolbuffers/protobuf/blob/bd00671b924310c0353a730bf8fa77c44e0a9c72/src/google/protobuf/io/coded_stream.h#L1300-L1306
+        if *value < 0 {
+            return 10
+        }
```

Comment:
A side note: length computation logic is duplicated between this function and call to `i32::len_varint()`. That code needs clean up (deduplication), but it may be not trivial, and I will do it later.

Reply:
👌